html/js/tree/Node.js

```diff
@@ -0,0 +1,94 @@
+GAMBIT.TREE = (function (parentModule) {
+    "use strict";
+
+    /**
+    * Creates a new Node.
+    * @class
+    * @param {Node} [father] Father node. If null, this is root.
+    */
+    function Node(father) {
+        this.father = father;
+        this.children = [];
+
+        if (father === null) { // If this is root set level to 0
+            this.level = 0;
+        } else {
+            father.addChild(this);
+            this.level = father.level + 1;
+        }
+    }
+
+    /**
+    * Function that draws the node in the global canvas
+    */
+    Node.prototype.draw = function () {
+        var thisNode = this;
+        var circle = GAMBIT.canvas.circle(GAMBIT.CONSTANTS.CIRCLE_SIZE)
+            .y(this.y)
+            .x(this.x)
+            .mouseover(function() {
+                this.fill({ color: '#f06' });
+            })
+            .mouseout(function() {
+                this.fill({ color: '#000' });
+            })
+            .click(function() {
+                if (GAMBIT.MODE_ADD){
+                    if (thisNode.children.length === 0) {
+                        // Always start with two nodes
+                        GAMBIT.tree.addChildNodeTo(thisNode);
+                    }
+                    GAMBIT.tree.addChildNodeTo(thisNode);
+                    // Tell the tree to redraw itself
+                } else {
+                    GAMBIT.tree.deleteNode(thisNode);
+                }
+                GAMBIT.tree.draw();
+            });
+        // console.log("Drawing at y " + this.level*50 + " and x " + this.x);
+    };
+
+    /**
+    * Function that adds child to node
+    * @param {Node} node Node to add as child
+    */
+    Node.prototype.addChild = function (node) {
+        this.children.push(node);
+    };
+
+    /**
+    * Function that removes child node from children
+    * @param {Node} node Child nod to remove
+    */
+    Node.prototype.deleteChild = function (nodeToDelete) {
+        var indexInList = this.children.indexOf(nodeToDelete);
+		if (indexInList > -1) {
+			this.children.splice(indexInList, 1);
+		}
+    };
+
+    /**
+    * Function that finds out if node is leaf
+    * @return {Boolean} True if is leaf.
+    */
+    Node.prototype.isLeaf = function () {
+        if (this.children.length === 0) {
+            return true;
+        }
+        return false;
+    };
+
+    /**
+    * Function that changes node's father to a given one
+    * @param {Node} newFather New father for node
+    */
+    Node.prototype.changeFather = function (newFather) {
+        this.father = newFather;
+        this.father.addChild(this);
+    };
```

Comment:
Here I believe you forgot to remove the current node from its parents children list.

Also you have to check for null in two places:
1. If you end up checking the old parent's child list, then check this.father (parent) for null before.
2. In any case, here newFather (newParent) could also be null, we shoudl check it and either allow it, but skip the this.father.addChild(), or forbid it and throw an exception